lib/OpenQA/Git.pm

```diff
@@ -43,23 +46,26 @@ sub _validate_attributes ($self) {
     }
 }
 
+sub _run_cmd ($self, $args = undef) {
+    my @git = $self->_prepare_git_command();
+    return run_cmd_with_log_return_error([@git, @$args]);
+}
```

Comment:
It makes no sense to assume `undef` as default for `$args` when the subsequent code would print a warning when `$args` is `undef`. I suppose creating the temporary array `@git` can also be avoided.

```suggestion
sub _run_cmd ($self, $args) { run_cmd_with_log_return_error([$self->_prepare_git_command, @$args]) }
```